### ISSUE TITLE
[audio] Compare Sidon and MossFormer2_SE_48K denoisers

### ISSUE THREAD

**Issue by 25d-228** [25d-228/meiki-decks#92]
## Outcome

Download and prepare the official Sidon and MossFormer2_SE_48K denoisers through a local relay, compare them on the same Japanese and Korean VoxCPM2 outputs, and select exactly one denoiser for the later full regeneration of all completed Japanese and Korean audio.

This issue selects the denoiser only. It does not bulk-regenerate audio, modify cards, rebuild archives, or publish bundles.

## Current state

- PR #91 merged Korean 05 source at `fea5ccc8185ca1eea19c99de4b8fd0f7b23525d6`.
- Japanese 00–05 contain 9,700 cards.
- Korean 00–05 contain 9,780 cards.
- Existing generated audio is superseded because human listening found excessive background noise.
- The human prepared new denoised Japanese and Korean reference audio and a replacement server environment.
- Neither denoiser is downloaded or prepared on the server.
- The server may not download the model artifacts directly, so the executor must use the local computer as a relay.
- The next bulk-regeneration issue must use the winner from this comparison and must not open until this issue merges.

## Approved candidate identities

Use only these official upstream implementations.

### Sidon

```text
Code:       https://github.com/sarulab-speech/Sidon
Checkpoint: https://huggingface.co/sarulab-speech/sidon-v0.1
Variant:    official TorchScript Sidon v0.1 single-speaker restoration model
License:    MIT
```

Do not use Core ML, ONNX, GGUF, third-party quantized, or DialogueSidon variants.

### MossFormer2_SE_48K

```text
Code:       https://github.com/modelscope/ClearerVoice-Studio
Checkpoint: https://huggingface.co/alibabasglab/MossFormer2_SE_48K
Model name: MossFormer2_SE_48K
License:    Apache-2.0
```

Do not use an MLX conversion, quantized conversion, Replicate wrapper, ComfyUI plugin, or another speech-enhancement model.

Before download, record the exact Git commit and Hugging Face revision selected for each candidate. Use immutable revisions for the comparison.

## Local relay acquisition

Use this local relay root:

```text
~/Downloads/meiki-denoiser-model-relay/
```

Create one directory per candidate:

```text
~/Downloads/meiki-denoiser-model-relay/sidon/
~/Downloads/meiki-denoiser-model-relay/mossformer2-se-48k/
```

For each candidate:

1. Download the official code repository at the selected commit.
2. Download the official model snapshot at the selected immutable revision.
3. Record the repository commit, model revision, file inventory, total byte size, and SHA-256 for every model artifact.
4. Verify that no HTML error page, pointer stub, partial file, or missing large-file object was downloaded.
5. Transfer the exact downloaded files to the server.
6. Verify the server-side SHA-256 values against the local relay values before inference.

Use the existing Hugging Face command or Python client when available. If no Hugging Face download client exists locally, creating one isolated local virtual environment and installing only `huggingface_hub` for this relay is approved.

Prefer the existing server transfer method. When a direct file-transfer command is needed, use `rsync --archive --partial --checksum`; if `rsync` is unavailable, use `scp -r`. Do not install a transfer tool.

Do not download unrelated checkpoints, training datasets, demonstrations, quantizations, or alternate variants.

## Server preparation

Read and follow the server documentation before creating directories or environments.

Prepare separate isolated runtime environments for Sidon and MossFormer2_SE_48K. Installing only the dependencies declared by the pinned official upstream repositories is approved for this issue.

Rules:

- do not use `sudo`;
- do not modify the system Python or global package environment;
- do not upgrade the existing VoxCPM2 environment;
- do not share one environment when the candidates require incompatible Python or package versions;
- do not compile or train a model;
- do not download model files again from the server;
- do not substitute another package, checkpoint, conversion, or wrapper;
- stop if the required Python, CUDA, system library, or upstream inference entrypoint is unavailable without a system-level change.

Record:

- environment path;
- Python version;
- package lock or installed-package inventory;
- exact inference command;
- fixed parameters;
- input and output formats;
- model load and one-file smoke-test result.

Use the official Sidon v0.1 single-speaker restoration inference path and the official ClearerVoice `MossFormer2_SE_48K` speech-enhancement path.

## Fixed bilingual sample set

Use these existing card sentences:

```text
Japanese
ja-01-629   question
ja-05-2920  number/interrogative
ja-05-2422  formal/long
ja-04-1523  casual statement
ja-05-1454  long connected delivery

Korean
ko-04-1411  question
ko-04-2247  number/counter
ko-04-2242  ordinary polite speech
ko-04-1521  formal presentation
ko-04-0885  long connected delivery
```

Read each complete `sentence` from current `main`. Do not edit source text.

Use the new human-prepared Japanese and Korean reference pairs. Record each reference audio SHA-256 and exact transcript path. Do not use the previous reference audio.

Generate each untreated baseline once with the existing VoxCPM2 Ultimate setup. Feed the exact same untreated baseline to both denoisers.

Prepare lossless 48 kHz mono WAV comparison files. Preserve raw candidate outputs when their native format differs. Do not normalize, compress, trim, or otherwise improve one candidate differently from the other.

## Blind comparison layout

Create this ignored workspace:

```text
work/denoiser-comparison/
```

For each of the ten samples, retain:

```text
baseline.wav
A.wav
B.wav
```

Use one consistent blind A/B mapping for all samples. Store the mapping locally in:

```text
work/denoiser-comparison/key.json
```

Do not reveal the mapping before the human chooses `A`, `B`, or `neither`.

Copy the thirty listening files and a concise sample manifest to:

```text
~/Downloads/meiki-denoiser-comparison/
```

The manifest may identify language, card ID, and sample category. It must not reveal the A/B model mapping.

## Objective validation

For every baseline and candidate output, verify with existing server tools:

- file exists;
- file is nonempty;
- FFmpeg or FFprobe decodes the complete file;
- duration is positive;
- standardized listening output is 48 kHz mono;
- candidate duration differs from its baseline by no more than 100 ms or 1%, whichever is larger;
- no NaN or infinite samples exist;
- no clipped samples are introduced.

Record duration, sample rate, channels, peak level, and overall RMS level. These measurements describe the files; they do not replace listening judgment.

Do not add ASR, forced alignment, speech-quality scoring, source-separation metrics, a benchmark framework, or another dependency.

## Human selection

After objective checks pass, stop and report the human-accessible comparison directory.

Human selection is blocking. The human will choose:

```text
A
B
neither
```

The selected model must reduce background noise while preserving:

- the fixed female voice identity;
- Japanese and Korean pronunciation;
- sentence rhythm and prosody;
- consonant attacks and word endings;
- stable volume;
- absence of metallic, watery, pumping, warbling, or clipped artifacts.

Use one denoiser for both languages. If the human chooses `neither`, keep this issue open and report the blocker. Do not open bulk regeneration.

After the human chooses A or B:

1. reveal the local mapping;
2. record the selected denoiser in a concise `README.md` audio section;
3. record the exact code commit, model revision, commands, parameters, input/output format, and comparison issue;
4. state that the selected model applies to the next Japanese and Korean bulk-regeneration issue;
5. run repository-native tests, lint, and build;
6. create one branch and one draft pull request containing exactly one `Fixes #92`.

## Process details

No nested or in-process subagents are required.

If one independent manifest or result audit is useful, use at most one sequential `codex exec --ephemeral` process with:

```text
model: gpt-5.6-sol
reasoning effort: high
sandbox: read-only
output: JSON only
```

Do not resume or reuse that process. It must not edit the repository, model relay, server environment, or comparison files.

## Scope

Allowed tracked change after human selection:

```text
README.md
```

Use the pull-request body for candidate versions, code commits, model revisions, model-file hashes, commands, objective results, reference hashes, sample inventory, A/B mapping after selection, and the human decision.

Generated baselines, denoised outputs, manifests, keys, downloaded code, models, environments, references, and caches remain untracked.

## Preserve

- All Japanese and Korean card and coverage source.
- The existing VoxCPM2 generation runtime.
- The human-prepared reference audio and transcripts.
- Existing archive format and command interfaces.
- All unrelated generated files.

## Exclusions

Do not:

- bulk-regenerate Japanese or Korean audio;
- modify any card sentence, cloze, answer, coverage file, stage, deck, or bundle;
- rebuild a Japanese or Korean archive;
- open the bulk-regeneration issue;
- open Korean 06;
- add a denoiser abstraction, provider interface, queue, service, worker, benchmark framework, or review application;
- train or fine-tune either denoiser;
- download alternate variants or unrelated models;
- commit audio, models, environments, references, caches, relay files, or comparison artifacts;
- publish or upload an archive.

## Execution rules

Read and follow all applicable `AGENTS.md` files before changing anything. Issue-specific requirements override repository rules. The current `ORCHESTRATOR → EXECUTOR` or `HUMAN → EXECUTOR` message can authorize a specific exception for the current task. For details not specified by these sources, follow the surrounding repository conventions.

Do not create a branch or pull request before the blocking human selection.

After human selection, use one branch from current `main` and one pull request. The pull-request description must contain exactly one:

```text
Fixes #92
```

Requested changes remain on the same branch and pull request.

## Evidence and validation

Before human selection, report:

- exact candidate identities and licenses;
- exact code commits and model revisions;
- local and server model-file hashes;
- relay and transfer commands;
- isolated environment details;
- exact inference commands and parameters;
- new reference SHA-256 values;
- ten-sample manifest;
- objective validation table;
- human-accessible comparison path;
- any candidate failure.

After human selection:

- update `README.md` with the selected denoiser;
- run repository-native tests, lint, and build;
- open one draft pull request;
- let CI provide the authoritative repository checks.

No automated test is required for subjective denoising quality. Objective file validation and one blinded human comparison are the required evidence.

## Tools, dependencies, and blockers

Approved downloads and tools for this issue are limited to:

- the two official code repositories named above;
- the two official model repositories named above;
- dependencies declared by the pinned official repositories, installed only in isolated environments;
- an isolated local `huggingface_hub` relay environment when required;
- existing Git, GitHub, Hugging Face, rsync or scp, Python environment, VoxCPM2, FFmpeg, FFprobe, shell, and repository tools.

Stop before downloading or installing anything outside this list.

Stop rather than guess when a revision, reference path, server path, command, format, dependency, CUDA requirement, or upstream inference entrypoint is unclear.

## External or interactive actions

Authorized for this issue:

- download the two named official code repositories and model snapshots to the local relay;
- create one isolated local download environment when required;
- transfer the downloaded files to the prepared server;
- create isolated server runtime environments using only pinned upstream dependencies;
- run the existing VoxCPM2 setup and the two denoisers on the prepared server;
- create comparison files under ignored `work/denoiser-comparison/`;
- copy the fixed listening set to `~/Downloads/meiki-denoiser-comparison/`;
- after human selection, create and push one branch and open or update one draft pull request.

No graphical control, system-wide installation, `sudo`, deployment, service, publication, archive upload, system-setting change, or other external action is authorized.

## File-retention rule

Do not delete files after each download, transfer, sample, candidate, or validation pass. Do not request per-file deletion permission.

Keep all relay files, code checkouts, model snapshots, isolated environments, baselines, raw outputs, standardized outputs, manifests, and the A/B key until the human selection and pull-request work are complete.

At final cleanup only:

- first measure the local relay, server candidate workspaces, and available disk space;
- if their combined transient size is at most 20 GiB and each relevant filesystem has at least 15% free space, leave all ignored files in place;
- otherwise delete only the losing candidate's duplicate relay/server model copy and disposable temporary conversion files;
- keep the selected denoiser on the server for the next bulk-regeneration issue;
- keep the human-prepared reference audio and transcripts;
- do not delete the blind comparison set before issue #92 merges;
- perform at most one cleanup operation at the end, not repeated cleanup during the comparison.

## Completion criteria

- Both official candidates are downloaded through the local relay and verified on the server.
- Both isolated runtimes execute their approved official model variant.
- Both candidates process the same ten untreated baselines.
- All thirty listening files pass objective validation.
- The human chooses A or B; `neither` remains a blocker.
- The selected mapping is recorded.
- `README.md` records the exact selected denoiser and fixed invocation.
- Repository checks and CI pass.
- The pull-request description contains exactly one `Fixes #92`.
- No audio, model, environment, reference, cache, relay, or comparison artifact is committed.
- Bulk regeneration remains unopened.
- No requirement remains uncovered, or each uncovered requirement is reported.

## Executor response protocol

Follow the root `AGENTS.md` response protocol.
